backend/src/modules/player-profiles/presentation/rest-api/request/PostPlayerProfileRequestBody.ts

```diff
@@ -0,0 +1,10 @@
+import { PlayerId } from '../../../../../shared/core/domain/PlayerId';
+
+export class PostPlayerProfileRequestBody {
+  //TODO playerId here should be string ? so also in CreatePlayerProfile it should be string, but than there is problem in CommandHandler :-/
```

Comment:
@nowakprojects tutaj mam mały problem z tym typem w PlayerId ... bo jak dam tu na string to to się ciągnie do commandHandlera gdzie potem przekazuję do domain command. Jeżeli w command CreatePlayerProfile zmienię playerId na string (tak jest w tournaments registrations) to wtedy w tym command ze screena muszę jako podmienić tym playerId ze stringa na type PlayerId, ale to jest readonly. Hmmm prosiłbym o podpowiedź :-) (późno jest, więc może jutro jakoś sam dojdę do tego, ale wolę na zaś zapytać) ... bez tego nie działa mi metoda w Swagger, bo nie widzi playerId :-P
![image](https://user-images.githubusercontent.com/31566345/109440195-9bc07680-7a31-11eb-89a3-cff18fd0d3ae.png)

Reply:
@Szambelan coś mi nie pasuje w tym rozwiązaniu, żeby w request body nadawać polu playerId typ PlayerId, no bo załóżmy ta chwilę taki scenariusz, że obsługujący aplikację tworzy nowego zawodnika i ma możliwość nadania mu id jakiego chce, to teraz przy takiej implementacji użytkownik musiałby wiedzieć jaki to jest typ ten PlayerId, żeby wysłać poprawne zapytanie POST. a druga sprawa jest taka - czy wiesz już gdzie chcesz gernerować ten id tego gracza? bo może to się zadziać w routerze tak jak Matuesz miał na początku zaimplementowane w open tournaments registrations albo możesz to zrobić przy zapisie do bazy

Reply:
Bardzo dobrze masz typy :) PlayerId to jest typ domenowy- ValueObject, więc jest tworzony z tego co przychodzi w commanda w commandHandlerze. Przemapuj sobie obiekt, żeby się zgadzał do sygnatury metody :) Przecież to przepisanie zmiennych.

Reply:
AAaa nie zauważyłem. Teraz po komenatrzu @tomdworniczak zwróciłem uwagę na PostPlayerProfileRequestBody.
To ma rację. W requescie muszą przychodzić zwykłe stringi :) W momencie wysyłania requesta nie ma jak zweryfikować czy to jest typ PlayerId. Dodatkowo musiałbyś to przekazywać jako `playerId: {raw: "asdasd"}`. To jest ValueObject, czyli pojawia się dopiero przy interakcji z domeną. Teraz do wartswy prezerntacji leci obiekt domenowy i wszelka zmiana w nim skutkowałaby zmianą w REST API i vice-versa. Czego chcemy uniknąć wproawdzająca warstawy.

Reply:
cholera usunąłem ten komentarz xD

Reply:
> Bardzo dobrze masz typy :) PlayerId to jest typ domenowy- ValueObject (domena może zdecydować czy odrzuca coś co przyszło z commanda, czy nie), więc jest tworzony z tego co przychodzi w commanda w commandHandlerze. Przemapuj sobie obiekt, żeby się zgadzał do sygnatury metody :) Przecież to przepisanie zmiennych.

no tak tak, jak zmienię na stringi w RequestBody to wtedy to muszę przemapować i będzie git. Myślałem, że da się to mapowanie jakoś ominąć... już poprawiam ... o 2 w nocy średnio mi szło myślenie, czy mapowanie to jedyna opcja, czy nie :-P

Reply:
done

Reply:
Dałoby radę, ale wtedy to się mija z celem. Bo my nie chcemy, żeby REST API było świadome tego jak wygląda domena i vice versa - żeby zmiany w REST API / domenie wpływały na drugie. A używając tego samego typu wpraowdzamy takie powiązanie.